changelog.md

```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## NEXT
+
+- Added `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.add_node_to_cache` and
+  `cai_causal_graph.time_series_causal_graph.TimeSeriesCausalGraph.remove_node_from_cache` to avoid having to loop
+  through the nodes of the graph multiple times.
```

Comment:
Once these methods are protected, I would just mention that we've improved the efficiency of certain methods, and list the methods that have been improved. The user doesn't need to know about the specifics.